kfs-purap/src/main/java/edu/arizona/kfs/module/purap/document/PurchaseOrderRetransmitDocument.java

```diff
@@ -0,0 +1,204 @@
+            if (item.isItemSelectedForRetransmitIndicator()) {
+                KualiDecimal taxAmount = item.getItemTaxAmount();
+                KualiDecimal itemTotal = (taxAmount != null) ? taxAmount : KualiDecimal.ZERO;
+                total = total.add(itemTotal);
+            }
+        }
+
+        return total;
+    }
+
+    /**
+     * When Purchase Order Retransmit document has been Processed through Workflow, the PO status remains to "OPEN" and the last
+     * transmit date is updated.
+     *
+     * @see org.kuali.kfs.module.purap.document.PurchaseOrderDocument#doRouteStatusChange()
+     */
+    @Override
+    public void doRouteStatusChange(DocumentRouteStatusChange statusChangeEvent) {
+        super.doRouteStatusChange(statusChangeEvent);
+
+        try {
+            // DOCUMENT PROCESSED
+            if (this.getFinancialSystemDocumentHeader().getWorkflowDocument().isProcessed()) {
+                SpringContext.getBean(PurchaseOrderService.class).setCurrentAndPendingIndicatorsForApprovedPODocuments(this);
+                setPurchaseOrderLastTransmitTimestamp(SpringContext.getBean(DateTimeService.class).getCurrentTimestamp());
+                updateAndSaveAppDocStatus(PurchaseOrderStatuses.APPDOC_OPEN);
+            }
+            // DOCUMENT DISAPPROVED
+            else if (this.getFinancialSystemDocumentHeader().getWorkflowDocument().isDisapproved()) {
+                SpringContext.getBean(PurchaseOrderService.class).setCurrentAndPendingIndicatorsForDisapprovedChangePODocuments(this);
+
+                // for app doc status
+                try {
+                    String nodeName = SpringContext.getBean(WorkflowDocumentService.class).getCurrentRouteLevelName(this.getFinancialSystemDocumentHeader().getWorkflowDocument());
+                    String reqStatus = PurapConstants.PurchaseOrderStatuses.getPurchaseOrderAppDocDisapproveStatuses().get(nodeName);
+                    updateAndSaveAppDocStatus(PurapConstants.PurchaseOrderStatuses.getPurchaseOrderAppDocDisapproveStatuses().get(reqStatus));
+                } catch (WorkflowException e) {
+                    logAndThrowRuntimeException("Error saving routing data while saving App Doc Status " + getDocumentNumber(), e);
+                }
+            }
+            // DOCUMENT CANCELED
+            else if (this.getFinancialSystemDocumentHeader().getWorkflowDocument().isCanceled()) {
+                SpringContext.getBean(PurchaseOrderService.class).setCurrentAndPendingIndicatorsForCancelledChangePODocuments(this);
+                // for app doc status
+                updateAndSaveAppDocStatus(PurapConstants.PurchaseOrderStatuses.APPDOC_CANCELLED);
+            }
+        }
+        catch (WorkflowException e) {
+            logAndThrowRuntimeException("Error saving routing data while saving document with id " + getDocumentNumber(), e);
+        }
+    }
+
+    public boolean isShouldDisplayRetransmitTab() {
```

Comment:
If you extend the correct parent class, this getter/setter pair of methods is not needed.